File: original_recipe/bbcursive/src/main/java/bbcursive/lib/abort.java
package bbcursive.lib;

import bbcursive.std;
import org.jetbrains.annotations.NotNull;

import java.nio.ByteBuffer;
import java.util.function.UnaryOperator;

/**
 * Created by jim on 1/17/16.
 */
public class abort {
    @NotNull
    public static UnaryOperator<ByteBuffer> abort(int rollbackPosition) {
        return b -> null == b ? null : std.bb(b, pos.pos(rollbackPosition), null);
    }
}

File: original_recipe/bbcursive/src/main/java/bbcursive/lib/abort.kt
package bbcursive.lib

import bbcursive.std
import java.nio.ByteBuffer
import kotlin.Int

/**
 * Created by jim on 1/17/16.
 */
object abort {
    @JvmStatic
    fun abort(rollbackPosition: Int) ={it:ByteBuffer?->
         if (null == it) null else std.bb(it,
            pos.pos(rollbackPosition),
            null)
    }
}